### ISSUE TITLE
Serialization / pickling

### ISSUE THREAD

**Issue by mishaturnbull** [mishaturnbull/edgegraph#66]
Add support for *all* edgegraph objects, starting with `BaseObject`, to natively support serialization and/or pickling to and from a file.  The goal of this ticket is to be able to save an entire universe to a file, load it back in from nothing, and continue processing as if nothing had happened.

**Must-haves**:
* [x] Ability to save Vertex and Link objects to a file, including all defined-during-runtime attributes
* [x] Ability to load Vertex and Link objects to a file, including all defined-during-runtime attributes
* [x] Save/load functionality must pass to all subclasses of Vertex and Link, including user-defined subclasses
* [x] Full singleton and semi-singleton support
* [x] Accurate reconstruction of graphs (Universes)
* [x] Arbitrary executable code **support** -- to include a big red safety banner in documentation -- for supporting future executable state machines / FAs

**Nice-to-haves**:
* [x] Save/load functionality defined at the BaseObject level, instead of Vertex / Link layer (N/A, applies by default)
* [ ] Human-readable-***ish*** format (JSON, XML, whatever; it doesn't have to be *fun* to edit, just *possible*)
* [ ] Option to save as uncompressed human-readable format and compressed binary format (zip, native pickle protocol, etc)

**Comment by mishaturnbull** [comment]
Should probably be done after #26.
